syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package iot_service;

// The temperature service definition.
service IoTService {
  // Responds with a temperature measurement
  rpc SayTemperature (TemperatureRequest) returns (TemperatureReply) {}
  
  // Send a command to the led
  rpc BlinkLed (LedRequest) returns (LedReply) {}

  // Responds with the current reading of a given light sensor
  rpc SayLightLevel (LightLevelRequest) returns (LightLevelReply) {}

  rpc Login(LoginRequest) returns (LoginReply) {}
}

// The request message containing the user's name.
message TemperatureRequest {
  string sensorId = 1;
}

// The response message containing the greetings
message TemperatureReply {
  string temperature = 1;
}

message LedRequest {
  uint32 state = 1;
  string ledId = 2;
}

message LedReply {
  uint32 ledstate = 1;
}

message LightLevelRequest {
  string sensorId = 1;
}

message LightLevelReply {
  string lightLevel = 1;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginReply {
  string reply = 1;
}